shared/connections.go

```diff
@@ -0,0 +1,64 @@
+package shared
+
+import (
+	"bufio"
+	"log"
+	"os"
+	"strings"
+)
+
+type NodeRole string
+
+const (
+	RoleMaster  NodeRole = "Master"
```

Comment:
All caps for enum?

Reply:
Go syntax is upper case = public, lower case = private (it's stupid I know)

Reply:
or well, RoleMaster -> ROLE_MASTER

Reply:
oh sure